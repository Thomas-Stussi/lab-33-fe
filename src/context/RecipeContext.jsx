/* eslint-disable keyword-spacing */
import React, { useState } from 'react';
import { recipeFetch } from '../services/historicalRecipesApi';

export const RecipeContext = React.createContext();

// eslint-disable-next-line react/prop-types
export const RecipeProvider = ({ children }) => {
  const [name, setName] = useState('');
  const [ingredients, setIngredients] = useState('');
  const [directions, setDirections] = useState('');
  const [source, setSource] = useState('');
  const [date, setDate] = useState('');
  const [method, setMethod] = useState('GET');
  const [id, setId] = useState(null);
  const [display, setDisplay] = useState({ 'Please': 'make a Request' });

  const handleChange = ({ target }) => {
    if (target.name === 'name') setName(target.value);
    if (target.name === 'ingredients') setIngredients(target.value);
    if (target.name === 'directions') setDirections(target.value);
    if (target.name === 'source') setSource(target.value);
    if (target.name === 'date') setDate(target.value);
    if (target.name === 'method') setMethod(target.value);
    if (target.name === 'id') setId(target.value);
  };

  const body = {
    name,
    ingredients,
    directions,
    source,
    date,
  };

  const handleSubmit = event => {
    event.preventDefault();

    return recipeFetch(method, JSON.stringify(body), id)
      .then(res => setDisplay(res));
  };

  const recipeArray = () => {
    return recipeFetch();
  };

  const bucket = {
    name,
    ingredients,
    directions,
    source,
    date,
    handleChange,
    handleSubmit,
    recipeArray,
    method,
    body,
    display
  };

  return (
    <RecipeContext.Provider value={bucket}>
      {children}
    </RecipeContext.Provider>
  );
};
