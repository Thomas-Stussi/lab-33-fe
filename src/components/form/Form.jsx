/* eslint-disable max-len */
import React, { useContext } from 'react';
import { RecipeContext } from '../../context/RecipeContext';
import styles from './Form.css';

// eslint-disable-next-line react/prop-types
const Form = () => {
  const { handleSubmit, handleChange } = useContext(RecipeContext);
  return (
    <>
      <form onSubmit={handleSubmit} className={styles.form}>
        <h1>Welcome to Historical Recipes, Use GET to see all!</h1>

        <section className={styles.form}>
          <label htmlFor="name">Name</label>
          <input id="name" name="name" onChange={handleChange} />

          <label htmlFor="ingredients" >Ingredients</label>
          <input id="ingredients" name="ingredients" onChange={handleChange} />

          <label htmlFor="directions" >Directions</label>
          <input id="directions" name="directions" onChange={handleChange} />

          <label htmlFor="source" >Source</label>
          <input id="source" name="source" onChange={handleChange} />

          <label htmlFor="date" >Date</label>
          <input id="date" name="date" onChange={handleChange} />

          <label htmlFor="id">ID</label>
          <input type="number" id="id" name="id" data-testid="id" onChange={handleChange} />
        </section>

        <section>
          <input type="radio" id="get" name="method" value="GET" data-testid="get" onChange={handleChange} />
          <label htmlFor="get">GET</label>

          <input type="radio" id="post" name="method" value="POST" onChange={handleChange} />
          <label htmlFor="post">POST</label>

          <input type="radio" id="put" name="method" value="PUT" onChange={handleChange} />
          <label htmlFor="put">PUT</label>

          <input type="radio" id="patch" name="method" value="PATCH" onChange={handleChange} />
          <label htmlFor="patch">PATCH</label>

          <input type="radio" id="delete" name="method" value="DELETE" onChange={handleChange} />
          <label htmlFor="delete">DELETE</label>
        </section>

        <button>Create Fetch!</button>
      </form>
    </>
  );
};

export default Form;
