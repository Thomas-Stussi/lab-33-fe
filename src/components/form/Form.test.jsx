import React, { useContext } from 'react';
import { render } from '@testing-library/react';
import Form from './Form';
import { RecipeProvider } from '../../context/RecipeContext';

describe('<Form>', () => {
  it('renders the form', () => {
    const { asFragment } = render(
      <RecipeProvider>
        <Form />
      </RecipeProvider>
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
