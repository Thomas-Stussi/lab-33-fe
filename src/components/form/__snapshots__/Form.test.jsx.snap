// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Form> renders the form 1`] = `
<DocumentFragment>
  <form
    class="form"
  >
    <h1>
      Welcome to Historical Recipes, Use GET to see all!
    </h1>
    <section
      class="form"
    >
      <label
        for="name"
      >
        Name
      </label>
      <input
        id="name"
        name="name"
      />
      <label
        for="ingredients"
      >
        Ingredients
      </label>
      <input
        id="ingredients"
        name="ingredients"
      />
      <label
        for="directions"
      >
        Directions
      </label>
      <input
        id="directions"
        name="directions"
      />
      <label
        for="source"
      >
        Source
      </label>
      <input
        id="source"
        name="source"
      />
      <label
        for="date"
      >
        Date
      </label>
      <input
        id="date"
        name="date"
      />
      <label
        for="id"
      >
        ID
      </label>
      <input
        data-testid="id"
        id="id"
        name="id"
        type="number"
      />
    </section>
    <section>
      <input
        data-testid="get"
        id="get"
        name="method"
        type="radio"
        value="GET"
      />
      <label
        for="get"
      >
        GET
      </label>
      <input
        id="post"
        name="method"
        type="radio"
        value="POST"
      />
      <label
        for="post"
      >
        POST
      </label>
      <input
        id="put"
        name="method"
        type="radio"
        value="PUT"
      />
      <label
        for="put"
      >
        PUT
      </label>
      <input
        id="patch"
        name="method"
        type="radio"
        value="PATCH"
      />
      <label
        for="patch"
      >
        PATCH
      </label>
      <input
        id="delete"
        name="method"
        type="radio"
        value="DELETE"
      />
      <label
        for="delete"
      >
        DELETE
      </label>
    </section>
    <button>
      Create Fetch!
    </button>
  </form>
</DocumentFragment>
`;
